### DISCUSSION TITLE
RustDesk suspends my computer when the screen turns off even though it's configured not to do so

### DISCUSSION THREAD

**Discussion by vgcouso** [rustdesk/rustdesk#7570] [General]
### Bug Description

I have set up two different computers to turn off the screen after 10 minutes but not to suspend. 
What happens is that if I have RustDesk running, after a few hours the computers go into suspension. 
This doesn't happen if RustDesk is not running.

### How to Reproduce

Computers should be configured to turn off the screen after a certain period of time, but the sleep time should be set to never.
To know if it has been suspended, I check the power light and also on one of the computers I have a web server set up and it becomes inaccessible.

### Expected Behavior

The computers should turn off the screen but not go into suspension if it's not configured to do so

### Operating system(s) on local side and remote side

Windows 11 Pro 23H2 -> Windows 11 Pro 23H2

### RustDesk Version(s) on local side and remote side

1.2.3-1 -> 1.2.3-1

### Screenshots

![Energy Configuration](https://github.com/rustdesk/rustdesk/assets/7522471/e7a5f892-b423-4bb9-b242-63c42cd8039f)

**Comment by rustdesk** [comment]
What does "RustDesk is not running." mean? Open the GUI or keep background service running?

We do have code to `unsuspend` the system in some case, but no code to `suspend` the system intentionally, there may be some API causes this on some windows implicitly.

**Reply by vgcouso** [reply]
If the service and application are closed, the computer does not suspend when the screen turns off

**Comment by xlionjuan** [comment]
It may help you figure out why your pc going sleep
https://stackoverflow.com/questions/9035065/when-did-my-computer-went-to-sleep-the-last-time-window-7

**Comment by rustdesk** [comment]
@fufesou please test 1.2.3 and 1.2.4. We do call unsuspend in some case, but never call suspend explicitly. No idea where ruins system's suspend configurations, maybe wakelock's drop cause this.

**Reply by vgcouso** [reply]
It happens with the GUI closed and the service started. Also, in previous versions it didn't happen.

**Reply by vgcouso** [reply]
I will try versions 1.2.3 and 1.2.4. Where can I download them?

**Reply by rustdesk** [reply]
Thanks

**Reply by vgcouso** [reply]
I installed version 1.2.3 (2023-10-13) and I'm going to leave the computer to see if it suspends. I don't think version 1.2.4 has been released yet, right?

**Reply by rustdesk** [reply]
1.2.4 is the nightly build,  https://github.com/rustdesk/rustdesk/releases/tag/nightly
12.3 is almost the same to 1.2.3-1, but 1.2.4 does have a lot of changes.

**Reply by rustdesk** [reply]
>  in previous versions it didn't happen.

Means 1.2.2 or even older versions?

**Reply by vgcouso** [reply]
I don't remenber

**Reply by vgcouso** [reply]
With Version 1.2.3, it also suspended with the screen turned off. I'm going to try 1.2.4 and I'll let you know.

**Reply by rustdesk** [reply]
Thanks. 1.2.3 is expected to same to 1.2.3-1.

**Reply by rustdesk** [reply]
If possbile, you can also test 1.2.2 if 1.2.4 has the same problem. It will help to tell us when we start to have this problem.

**Comment by vgcouso** [comment]
ok, after 1.2.4

**Reply by vgcouso** [reply]
With Version 1.2.4, it also suspended with the screen turned off. I'm going to try 1.2.2 and I'll let you know.

**Reply by vgcouso** [reply]
With Version 1.2.2, the computer also suspends after the screen turns off. I will stop the RustDesk service to confirm the error.

**Comment by fufesou** [comment]
@vgcouso Hi, I've tried two PCs. Win11 and Win10, 1.2.3-1 and the nightly build.

I cannot reproduce this issue. My PCs do not sleep

I do the following steps. 

1. Installed RustDesk and have the client and service running.

2. Set the battery configurations ![image](https://github.com/rustdesk/rustdesk/assets/13586388/4db991c4-a27e-4b0f-a4d8-e98bdf51d693)

3. After the PCs have been left there for hours, they can still connect normally. 
Or after mouse hovering, the screen can be waked, the PC is still working.

> I have set up two different computers to turn off the screen after 10 minutes but not to suspend.

I'm assuming here that by suspend you mean sleep.
Does the computer need to be restarted, or can it be woken up by just pressing the keyboard?

**Comment by rustdesk** [comment]
Obviously, it is a rare case on specific machines, since 1.2.3 has been widely deployed.

**Reply by vgcouso** [reply]
I don't know what's happening. I've uninstalled it and it still sunspends. I'm going to format the computer and reinstall Windows to see if it's the operating system. If it's not the case, I'll reinstall RustDesk to confirm whether or not it's the cause of the problem. I'll keep you informed.

**Reply by vgcouso** [reply]
I've formatted the computer, reinstalled Windows, and only reinstalled RustDesk. The computer hasn't suspended with the screen off. I think it's something else. I'll check it out. I'm really sorry for creating this bug, but I was convinced it was RustDesk. I apologize to you all. Meanwhile, I'll continue trying to figure out what's causing the issue.
